aiida/orm/nodes/data/container_code.py

```diff
@@ -0,0 +1,84 @@
+# -*- coding: utf-8 -*-
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""Data plugin represeting a containter executable code to be wrapped and called through a `CalcJob` plugin."""
+import os
+
+from aiida.common import exceptions
+from aiida.common.log import override_log_level
+
+from .code import Code
+from .data import Data
+
+__all__ = ('ContainerCode',)
+
+
+class ContainerCode(Code):
```

Comment:
Honestly, I am not convinced anymore whether creating a new `ContainerCode` class is better than adding one more argument to the standard `Code`

Reply:
> Honestly, I am not convinced anymore whether creating a new ContainerCode class is better than adding a few more argument to the standard Code

Would you mind sharing why? That route has been tried multiple times (local and remote for `Code`, mesh and explicit for `KpointsData`) and it always creates a lot of confusion, both in the code and for the user.